Doc/library/contextlib.rst

```diff
@@ -126,6 +126,16 @@ Functions and classes provided:
 
    .. versionadded:: 3.7
 
+   :func:`asynccontextmanager` uses :class:`AsyncContextDecorator` so the context managers
+   it creates can be used as decorators as well as in :keyword:`async with` statements.
+   When used as a decorator, a new generator instance is implicitly created on
+   each function call (this allows the otherwise "one-shot" context managers
+   created by :func:`asynccontextmanager` to meet the requirement that context
+   managers support multiple invocations in order to be used as decorators).
+
+  .. versionchanged:: 3.10
+  Use of :class:`AsyncContextDecorator`.
```

Comment:
`AsyncContextDecorator` is a low-level implementation detail. It doesn't need to be exposed because it will confuse a regular Python user.  Here's how I'd rewrite the doc:

```suggestion
   Context managers defined with :func:`asynccontextmanager` can be used
   either as decorators or with :keyword:`async with` statements::
   
     import time
   
     async def timeit():         
         now = time.monotonic()
         try:
             yield
         finally:
             print(f'it took {time.monotonic() - now}s to run')
             
      @timeit()
      async def main():
          # ... async code ...

   When used as a decorator, a new generator instance is implicitly created on
   each function call. This allows the otherwise "one-shot" context managers
   created by :func:`asynccontextmanager` to meet the requirement that context
   managers support multiple invocations in order to be used as decorators.

  .. versionchanged:: 3.10
     Async context managers created with :func:`asynccontextmanager` can
     be used as decorators.
```

Reply:
This language was copied from contextmanager docs, so the same argument could be applied there.  The docs of both managers should be consistent.

Reply:
I'd say that the contextmanager docs should be updated to match my suggestion.

Reply:
But there's no "rule" that the docs for different (albeit similar) APIs must look the same.